nimbus/src/matcher.rs

```diff
@@ -29,9 +31,32 @@ pub struct Matcher {
     pub debug_tags: Vec<String>,
 }
 
+/// The `AppContext` object represents the parameters and characteristics of the
+/// consuming application that we are interested in for targeting purposes. The
+/// `app_name`, `app_id`, and `channel` fields are not optional as they are expected
+/// to be provided by all consuming applications as they are used in the first
+/// pieces of targeting that help to ensure that an experiment is only processed
+/// by the correct application.
+///
+/// Definitions of the fields are as follows:
+/// - `app_name`: This is the name of the application (e.g. "Fenix" or "Firefox iOS")
+/// - `app_id`: This is the application identifier, especially for mobile (e.g. "org.mozilla.fenix")
+/// - `channel`: This is the delivery channel of the application (e.g "nightly")
+/// - `app_version`: The user visible version string (e.g. "1.0.3")
+/// - `app_build`: The build identifier generated by the CI system (e.g. "1234/A")
+/// - `architecture`: The architecture of the device, (e.g. "arm", "x86")
+/// - `device_manufacturer`: The manufacturer of the device the application is running on
+/// - `device_model`: The model of the device the application is running on
+/// - `locale`: The locale of the application during initialization (e.g. "es-ES")
+/// - `os`: The name of the operating system (e.g. "Android", "iOS", "Darwin", "Windows")
+/// - `os_version`: The user-visible version of the operating system (e.g. "1.2.3")
+/// - `android_sdk_version`: Android specific for targeting specific sdk versions
+/// - `debug_tag`: Used for debug purposes as a way to match only developer builds, etc.
 #[derive(Deserialize, Serialize, Debug, Clone, Default)]
 pub struct AppContext {
+    pub app_name: String,
     pub app_id: String,
+    pub channel: String,
```

Comment:
These will have to be supplied by the consuming application.  `app_name` will match the naming changes that go into some of our scraping tools, namely probe-scraper, in order to make it easier to connect the data in the dashboards.  `channel` is sort of arbitrary on mobile, I agree, especially since `app_id` will be different per channel, but this is something that we will need to support for Desktop and something that could be supplied by mobile.

Reply:
Let me see if I can dig up a link to the doc about the naming conventions in the data tools that is what is driving some of this for more context...

Reply:
They'll need to come from the fenix build, so will have to passed into A/C from Fenix. This question shouldn't block this being landed, but I note we'll need to make A/C and Fenix changes. :(